src/main/java/uk/gov/hmcts/reform/slc/services/ReportParser.java

```diff
@@ -31,11 +30,14 @@ public ParsedReport parse(Report report) {
 
             List<LetterPrintStatus> statuses =
                 stream(parser.spliterator(), false)
-                    .map(row -> toPrintStatus(row))
-                    .filter(Objects::nonNull)
+                    .map(csvRecord -> toPrintStatus(csvRecord))
                     .collect(toList());
 
-            return new ParsedReport(report.path, statuses);
+            return new ParsedReport(
+                report.path,
+                statuses.stream().filter(s -> s != null).collect(toList()),
```

Comment:
It would be best to name the results of statuses.stream()... before passing them. Not strong about it.